### ISSUE TITLE
Editor issues removing task decomposition variables in offline mode

### ISSUE THREAD

**Issue by yawlfoundation** [yawlfoundation/yawl#385]
```
If the Editor is in "offline" mode (no Engine and Resource Service
connections), strange behaviour occurs when trying to remove task
decomposition variables.

The following are observed use cases (just open the Editor without the YAWL
backend having been started, create a new specification, add a new atomic
task and create a task decomposition for it):

1.
-- Add any two variables 
-- Remove the first; both get removed instead of only the first

This still occurs if you add the two, click Done, go back into the task
decomposition and then try to remove the first

2.
-- Add any two variables
-- Save the spec. and close it
-- Reopen the spec, edit the task decomposition and remove the first
-- Display doesn't update and can't click on the variable lines properly,
but clicking Done and going back in shows that the removal was actually
done properly

In response to previous bugs I logged, Michael changed the Editor for 2.0.1
to check for Engine/RS connections in certain situations. I suspect that
this code has introduced these problems when in offline mode (might be
wrong of course! May still have existed in v2.0).

Putting as priority Low since appears not to occur (at least not
consistently) when the Editor has Engine and RS connections.

What steps will reproduce the problem?
As above

What is the expected output? What do you see instead?
As above


```

Original issue reported on code.google.com by `monsieur...@gmail.com` on 31 Mar 2010 at 11:12

**Comment by yawlfoundation** [comment]
```
I experienced the same problems, but I not in off-line mode (i.e. I had the 
engine running and I had connection to the Resource Service). I am running the 
2.1Beta version.

Petia
```

Original comment by `petia.wo...@gmail.com` on 10 Jun 2010 at 9:37

**Comment by yawlfoundation** [comment]
```
fixed for 2.2 (r. 1771)
```

Original comment by `yawl.mic...@gmail.com` on 15 Aug 2011 at 3:49
* Changed state: **Fixed**
